src/Microsoft.PowerShell.ConsoleHost/host/msh/CommandLineParameterParser.cs

```diff
@@ -651,6 +652,25 @@ private static bool MatchSwitch(string switchKey, string match, string smallestU
                     && match.StartsWith(switchKey, StringComparison.OrdinalIgnoreCase));
         }
 
+        private static bool IsStaSupported()
+        {
+#if UNIX
+            return false;
+#else
+            Thread thread = new Thread(() => { });
+            thread.SetApartmentState(ApartmentState.STA);
```

Comment:
Do we need to actually start the thread?  I assume yes, but want to check.  There is also a `TrySetApartmentState` to avoid exception handling, but again I am assuming this particular problem requires actually running a thread.  If so please add a comment about the issue being solved (not simply a `platform not supported` problem).

Reply:
Unfortunately, the exception only occurs at thread start time.  `TrySetApartmentState()` would only fail on a running thread.  I can add a comment.